include/ck_tile/ops/gemm/pipeline/block_gemm_pipeline_agmem_bgmem_creg_universal_policy.hpp

```diff
@@ -0,0 +1,421 @@
+                           make_pass_through_transform(kK1)),
+                make_tuple(sequence<0, 1, 4, 2, 7>{}, sequence<5, 6, 3>{}),
+                make_tuple(sequence<1>{}, sequence<0>{}));
+
+            return b_lds_block_desc_n_k;
+        }
+    }
+
+    template <typename Problem>
+    CK_TILE_HOST_DEVICE static constexpr ck_tile::index_t GetSmemSizeA()
+    {
+        constexpr index_t smem_size_a = sizeof(typename Problem::ADataType) *
+                                        MakeALdsBlockDescriptor<Problem>().get_element_space_size();
+        return smem_size_a;
+    }
+
+    template <typename Problem>
+    CK_TILE_HOST_DEVICE static constexpr ck_tile::index_t GetSmemSizeB()
+    {
+        constexpr index_t smem_size_b = sizeof(typename Problem::BDataType) *
+                                        MakeBLdsBlockDescriptor<Problem>().get_element_space_size();
+        return smem_size_b;
+    }
+
+    template <typename Problem>
+    CK_TILE_HOST_DEVICE static constexpr ck_tile::index_t GetSmemSize()
+    {
+        constexpr index_t smem_size_a = GetSmemSizeA<Problem>();
+        constexpr index_t smem_size_b = GetSmemSizeB<Problem>();
+        index_t smem_size             = 0;
+        smem_size += smem_size_a + smem_size_b;
+
+        return smem_size;
+    }
+
+    template <typename Problem>
+    CK_TILE_HOST_DEVICE static constexpr auto MakeADramTileDistribution()
+    {
+        using WarpGemm = WarpGemmMfmaDispatcher<typename Problem::ADataType,
+                                                typename Problem::BDataType,
+                                                typename Problem::CDataType,
+                                                Problem::BlockGemmShape::WarpTile::at(I0),
+                                                Problem::BlockGemmShape::WarpTile::at(I1),
+                                                Problem::BlockGemmShape::WarpTile::at(I2),
+                                                true>;
+
+        constexpr index_t kMPerBlock = Problem::BlockGemmShape::kM;
+        constexpr index_t kKPerBlock = Problem::BlockGemmShape::kK;
+
+        constexpr auto kM0 = get_warp_size() * Problem::BlockGemmShape::BlockWarps::at(I0);
+        constexpr auto kM1 = kMPerBlock / kM0;
+
+        constexpr index_t kK1 = WarpGemm::kK;
+        constexpr index_t kK0 = kKPerBlock / kK1;
+
+        static_assert(kM1 != 0, "M1 is zero, which will lead to a division by zero error.");
+
+        return make_static_tile_distribution(
+            tile_distribution_encoding<sequence<1>,
+                                       tuple<sequence<kM0, kM1>, sequence<kK0, kK1>>,
+                                       tuple<sequence<1>, sequence<2>>,
+                                       tuple<sequence<0>, sequence<1>>,
+                                       sequence<1, 2>,
+                                       sequence<1, 0>>{});
+    }
```

Comment:
Is that correct ? `P { M0, K1 }`,  `Y { M1, K0 }` .... 1st dim of P should be # of waves and 2nd dim should be # of threads in wavefront....